beacon/validator/block_builder.go

```diff
@@ -335,11 +326,10 @@ func (s *Service[
 	body.SetDeposits(deposits)
 
 	var eth1Data *ctypes.Eth1Data
-	// TODO: assemble real eth1data.
 	body.SetEth1Data(eth1Data.New(
-		common.Root{},
-		0,
-		common.ExecutionHash{},
+		ctypes.Deposits(deposits).HashTreeRoot(),
```

Comment:
You set the `DepositRoot` to the hash of the terminal segment of the list of deposits, but you are not linking it to the hash of the past deposits, i.e. the previous `Eth1Data`.

Reply:
Now using combi(historicalRoot, blkDeposits.HTR)